src/python-common/ceph/deployment/service_spec.py

```diff
@@ -1095,3 +1113,104 @@ def get_port(self) -> int:
             return {'prometheus': 9095,
                     'node-exporter': 9100,
                     'grafana': 3000}[self.service_type]
+
+
+class SNMPGatewaySpec(ServiceSpec):
+
+    def __init__(self,
+                 service_type: str = 'snmp-gateway',
+                 snmp_version: Optional[str] = None,
+                 snmp_destination: Optional[str] = None,
+                 credentials: Optional[Dict[str, str]] = None,
+                 engine_id: Optional[str] = None,
+                 auth_protocol: Optional[str] = None,
+                 privacy_protocol: Optional[str] = None,
+                 placement: Optional[PlacementSpec] = None,
+                 unmanaged: bool = False,
+                 preview_only: bool = False,
+                 port: Optional[int] = 9464,
+                ):
+        assert service_type == 'snmp-gateway'
+        assert credentials is not None
+
+        super(SNMPGatewaySpec, self).__init__(
+            service_type,
+            placement=placement,
+            unmanaged=unmanaged,
+            preview_only=preview_only)
+
+        self.service_type = service_type
+        self.snmp_version = snmp_version
+        self.snmp_destination = snmp_destination
+        self.port = port
+        self.credentials = credentials
+        self.engine_id = engine_id
+        self.auth_protocol = auth_protocol
+        self.privacy_protocol = privacy_protocol
+
+    @property
+    def ports(self) -> List[int]:
+        return [self.port]
```

Comment:
ceph/deployment/service_spec.py: note: In member "ports" of class "SNMPGatewaySpec":
ceph/deployment/service_spec.py:1153: error: List item 0 has incompatible type "Optional[int]"; expected "int"